src/cli/commands/retrieve_metrics.go

```diff
@@ -133,3 +148,67 @@ func RetrieveMetrics(cliConnection api.Connection, appName, metricName string, s
 
 	return nil
 }
+
+func RetrieveAggregatedMetrics(cliConnection api.Connection, appName, metricName string, startTime, endTime int64, desc bool, writer io.Writer) error {
+
+	cfclient, err := api.NewCFClient(cliConnection)
+	if err != nil {
+		return err
+	}
+	err = cfclient.Configure(appName)
+	if err != nil {
+		return err
+	}
+
+	endpoint, err := api.GetEndpoint()
+	if err != nil {
+		return err
+	}
+	if endpoint.URL == "" {
+		return errors.New(ui.NoEndpoint)
+	}
+
+	apihelper := api.NewAPIHelper(endpoint, cfclient, os.Getenv("CF_TRACE"))
+	ui.SayMessage(ui.ShowAggregatedMetricsHint, appName)
+
+	table := ui.NewTable(writer, []string{"Metrics Name", "Value", "Timestamp"})
+	var (
+		page     uint64 = 1
+		next     bool   = true
+		noResult bool   = true
+		data     [][]string
+	)
+	for next {
+		next, data, err = apihelper.GetAggregatedMetrics(metricName, startTime, endTime, desc, page)
+		if err != nil {
+			return err
+		}
+
+		for _, row := range data {
+			table.Add(row)
+		}
+		if len(data) > 0 {
+			noResult = false
+			table.Print()
```

Comment:
why print table every time you got a page?

Reply:
after each table.Print(),   the rows are cleared  https://github.com/cloudfoundry-incubator/app-autoscaler-cli-plugin/blob/master/src/cli/ui/table.go#L52,  so table.Print() appends the pages in fact, no overlapping introduced.

Reply:
I would prefer to add all rows to table, and then print it out

Reply:
when we have hundred of metrics or more to return, we need to query API server several times to get all rows. In this case,  the end user may experience a long latency if we hold the print to the end.

Reply:
if you have this concern, then you probably will need to add more informational message to indicating the user you are retrieving more ....

Reply:
Will reduce the metric retrieve time range in another PR.